src/plugins/intel_cpu/src/nodes/mkldnn_def_conv_node.h

```diff
@@ -86,29 +85,74 @@ class MKLDNNDeformableConvolutionNode : public MKLDNNNode {
     InferenceEngine::Precision getRuntimePrecision() const override;
 
 private:
-    size_t group = 1;
-    bool with_bilinear_pad = false;
-    std::vector<ptrdiff_t> stride = {};
-    std::vector<ptrdiff_t> dilation = {};
-    std::vector<ptrdiff_t> paddingL = {};
-
-    int deformable_group = 1;
-
-    jit_def_conv_params jcp = {};
+    struct DefConvAttr {
+        size_t group = 1;
+        int deformable_group = 1;
+        bool with_bilinear_pad = false;
+        std::vector<ptrdiff_t> stride = {};
+        std::vector<ptrdiff_t> dilation = {};
+    } defConvAttr;
 
     std::vector<int> sampledCoordsVector;
     std::vector<float> interpWeightsVector;
 
-    std::shared_ptr<jit_uni_def_conv_kernel> def_conv_kernel = nullptr;
+    void prepareParams() override;
+    void updatePadding();
 
-    void prepareSamplingWeights(const std::vector<size_t>& src_strides, const float* offsets, const std::vector<size_t>& off_strides,
-                                const float* modulation = nullptr, const std::vector<size_t>& modulation_strides = {});
-
-    void executeReference(const float* src, const float* weights, float* dst, const std::vector<size_t>& src_strides,
-                          const std::vector<size_t>& wei_strides, const std::vector<size_t>& dst_strides);
-    void executeOptimized(const float* src, const float* weights, float* dst,
-                          const std::vector<size_t>& src_strides, const std::vector<size_t>& dst_strides);
+    std::vector<ptrdiff_t> paddingL = {};
+    void executeDynamicImpl(mkldnn::stream strm) override;
+    static constexpr size_t DATA_ID = 0;
+    static constexpr size_t OFF_ID = 1;
+    static constexpr size_t WEI_ID = 2;
+    static constexpr size_t MOD_ID = 3;
+    std::string errorPrefix;
+    class DefConvExecutor {
+        public:
+            DefConvExecutor(const DefConvAttr &defConvAttr,
+                                const std::vector<std::shared_ptr<BlockedMemoryDesc>> &descVector,
+                                const std::vector<ptrdiff_t> &padL);
+
+            // we need passing of reference to outer class to have access to preallocated sampledCoords and interpWeights
+            virtual void exec(MKLDNNDeformableConvolutionNode *node, const float* src, const float* offsets,
```

Comment:
For me is doesn't look like a good design to make exec() be dependent on node itself (moreover non const one). In fact currently executor changes state of the name - this is not appropriate behavior. 
Isn't it enough to move sampledCoordsVector and interpWeightsVector inside executor to rid of this dependency?

Reply:
Discussed offline